client/lib/url-search-query-state/index.js

```diff
@@ -0,0 +1,67 @@
+import page from 'page';
+import { Component, useEffect, useState, useCallback } from 'react';
+
+function updateUrlSearchQuery( queryName, queryValue ) {
+	if ( ! window ) {
```

Comment:
If `window` isn't defined, this will trigger a `ReferenceError`; how about we do it with:

```suggestion
	if ( typeof window === 'undefined' ) {
```